src/plugins/discover/public/migrate_state.ts

```diff
@@ -66,7 +66,7 @@ export function migrateUrlState(oldPath: string, newPath = '/'): string {
   let path = newPath;
   const pathPatterns = [
     {
-      pattern: '#/context/:indexPattern/:id\\?:appState?',
+      pattern: '#/context/:indexPattern/:id?:appState?',
```

Comment:
This works because the a route like `#/context/ip/abcd` results in `indexPattern = ip`, `id = a` and `appState = bcd`

But it looks like we dont need the `appState` so lets just change the paths to 

```
    {
      pattern: '#/context/:indexPattern/:id',
      extraState: { docView: 'context' },
      path: `context`,
    },
    {
      pattern: '#/doc/:indexPattern/:index',
      extraState: { docView: 'doc' },
      path: `doc`,
    },
```

Also you forgot to add the break statement after line 101

Reply:
I dont think we need to add the break statement on line 101? since it will return path anyway @ashwin-pc

Reply:
So the way a switch statement works is that once it matches a particular switch case, everything under that is executed until it hits a break or a default statement. So in this case, once the dog or context parts are matched, all commands under that are exterior until it hits the final break statement, which means that everything inside the discover path is also executed. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
Can you also remove the appstate from the doc route?

Reply:
fixed